addons/hr_expense/models/hr_expense.py

```diff
@@ -1169,18 +1174,15 @@ def _prepare_payment_vals(self):
         else:
             currency = self.expense_line_ids[0].currency_id
             amount = sum(self.expense_line_ids.mapped('total_amount'))
-        rate = amount / self.total_amount if self.total_amount else 0.0
         move_lines = []
         for expense in self.expense_line_ids:
-            # Due to rounding and conversion mismatch between vendor bills and payments, we have to force the computation into company account
-            amount_currency_diff = expense.total_amount_company if currency == expense.company_currency_id else expense.total_amount
-            last_expense_line = None # Used for rounding in currency issues
-            tax_data = self.env['account.tax']._compute_taxes([expense._convert_to_tax_base_line_dict()])
-            for base_line_data, update in tax_data['base_lines_to_update']:  # Add base lines
+            tax_data = self.env['account.tax']._compute_taxes([expense._convert_to_tax_base_line_dict(price_unit=expense.total_amount_company, currency=expense.company_currency_id)])
+            tax_data_currency = self.env['account.tax']._compute_taxes([expense._convert_to_tax_base_line_dict(price_unit=expense.total_amount, currency=expense.currency_id)])
```

Comment:
I still don't understand why this is giving the same results as an invoice knowing the invoice is only computing taxes in foreign currency :/

Reply:
Because the amount in company currency is recomputed and is 1 cent higher than what I want if I don't do that and compute it from converting with rate from amount_currency, or worse if I go the other way

All happening in the sync dynamic line hell
All I know is doing this works exactly how I want it to work